modules/gdscript/gdscript.cpp

```diff
@@ -2432,27 +2430,20 @@ void GDScriptLanguage::get_reserved_words(List<String> *p_words) const {
 		p_words->push_back(*w);
 		w++;
 	}
-
-	List<StringName> functions;
-	GDScriptUtilityFunctions::get_function_list(&functions);
-
-	for (const StringName &E : functions) {
-		p_words->push_back(String(E));
-	}
```

Comment:
What's the justification for removing utility functions from the reserved list with regards to this PR? Also the same question about `float`, `int`, and `bool`, with only `void` being kept out of those types.

Does this affect the highlighter somehow? And what else does this affect.

Reply:
> Does this affect the highlighter somehow?

No, the result should be the same, except for intentional changes.

> And what else does this affect.

`get_reserved_words()` is used for code highlighting and to check an identifier in [`EditorAutoloadSettings`](https://github.com/godotengine/godot/blob/19890614c6a78ec36030ce65c7da05f07fcdb9ed/editor/editor_autoload_settings.cpp#L144-L156). I've only left here the keywords listed in [`GDScriptTokenizer`](https://github.com/godotengine/godot/blob/19890614c6a78ec36030ce65c7da05f07fcdb9ed/modules/gdscript/gdscript_tokenizer.cpp#L44) (although [some of them](https://github.com/godotengine/godot/blob/19890614c6a78ec36030ce65c7da05f07fcdb9ed/modules/gdscript/gdscript_tokenizer.cpp#L185-L201) are valid identifiers).

> What's the justification for removing utility functions from the reserved list

Highlighter checks them [separately](https://github.com/godotengine/godot/blob/5052b202385224cf632d58deb717132c16ddc86b/modules/gdscript/editor/gdscript_highlighter.cpp#L704-L713), and utility function names are valid identifiers (`class_name range` is valid).

> Also the same question about `float`, `int`, and `bool`, with only `void` being kept out of those types.

`void` is in the tokenizer, the rest are not. Built-in types are taken into account both in the [highlighter](https://github.com/godotengine/godot/blob/5052b202385224cf632d58deb717132c16ddc86b/modules/gdscript/editor/gdscript_highlighter.cpp#L678-L689) and in [`EditorAutoloadSettings`](https://github.com/godotengine/godot/blob/19890614c6a78ec36030ce65c7da05f07fcdb9ed/editor/editor_autoload_settings.cpp#L124-L132).